pkg/ddc/vineyard/operations/base.go

```diff
@@ -39,21 +41,42 @@ func NewVineyardFileUtils(podNamePrefix string, port int32, replicas int32, name
 	}
 }
 
+func validateResourceName(name string) error {
+	validNameRegex := `^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
+	matched, err := regexp.MatchString(validNameRegex, name)
+	if err != nil {
+		return fmt.Errorf("failed to validate name: %v", err)
+	}
+	if !matched {
+		return fmt.Errorf("invalid Kubernetes naming: %s", name)
+	}
+	return nil
+}
+
 // Get summary info of the Vineyard Engine
 func (a VineyardFileUtils) ReportSummary() (summary []string, err error) {
 	var resp *http.Response
 	var body []byte
+	if err := validateResourceName(a.podNamePrefix); err != nil {
+		return nil, err
+	}
+	if err := validateResourceName(a.namespace); err != nil {
+		return nil, err
+	}
+	client := &http.Client{
+		Timeout: 20 * time.Second,
+	}
 	for i := int32(0); i < a.replicas; i++ {
 		url := fmt.Sprintf("http://%s-%d.%s-svc.%s.svc.cluster.local:%d/metrics", a.podNamePrefix, i, a.podNamePrefix, a.namespace, a.port)
```

Comment:
In a default setup, the DNS service suffix for a Kubernetes cluster is ".cluster.local." However, the value is customizable and can be configured during cluster initialization. Therefore, it's possible that in some Kubernetes clusters, the value is not "cluster.local". 

Given varying configurations across clusters, it is a good practice not to hard-code this DNS suffix in your application. Instead, you should generally use serviceName.namespace as the DNS name for lookups. The overhead expense of the DNS lookup is relatively low and won't significantly impact performance. This practice would render your application more robust and adaptable to different Kubernetes environments.

Reply:
Make sense to me. 

Sorry, I forgot to delete this one `.cluster.local`.